### ISSUE TITLE
Subtree Walk - possible to get unhandled error "Cannot read property 'length' of undefined"

### ISSUE THREAD

**Issue by smithkaylar** [markabrahams/node-net-snmp#182]
Trace log shows the error throwing at line 2644 in index.js - I believe this is a V3 permissions issue but I'm not 100% confident.

Using [this SNMP Simulator](https://www.ireasoning.com/snmpsimulator.shtml) to reproduce:

1. Open the simulator
2. Go to File > Open Sample Project
3. Using the data from the `context indexing` sample create a v3 session and then do session.subtree. The configs I am using are as follows (taken from the simulator agent config file for context indexing): `createV3Session('localhost', user: { authKey:"abc12345", authProtocol:2, level:3, name:"newUser", privKey:"abc12345", privProtocol:2 }, options: { port:1600, retries:0, sourcePort:602, version:3 }` and then with that session `session.subtree('1.3', (varbinds) => {...}, (error) => {...})`. When the subtree command is called the `Cannot read property 'length' of undefined"` error is thrown.
4. I can change the port to the multi-walk example port (1610), update the version to v3 for that in the simulator, and successfully get a subtree. I can also change to the multi-walk example and update the root oid to be 1.4, which correctly returns nothing without the uncaught error being thrown.

The easiest method I've found locally to prevent the error from being thrown is to add the following before the if statement at line 2644:

```
if (!varbinds.length) {
	req.doneCb(null);
	return;
}
```

Although this works, it seems like maybe there is some sort of error occurring that isn't being caught since there is no `EndOfMibView` oid in the returned varbinds. Would appreciate any insight!

**Comment by markabrahams** [comment]
Hi @smithkaylar - I tried the simulator and was unsuccessful in getting SNMPv3 working at all - admittedly I didn't spend a lot of time troubleshooting or learning the simulator.  I got SNMPv2 working OK in the simulator, and I could use the library's subtree successfully - as you did yourself.

It seems like we're more debugging the simulator here than the library.  The lack of an `EndOfMibView` could be an issue with the simulator scenario.

An initial look at your PR #184  for this, I think it should fix this condition without adversely affecting the walk behaviour in general.  I'll have a deeper look/test hopefully over the next day or so with a view to merging the PR.
